client/features/data-encryption.js

```diff
@@ -0,0 +1,49 @@
+import WebSocketAsPromised from 'websocket-as-promised';
+
+export const decryptEventPayloads = async (events, port) => {
+  const sock = new WebSocketAsPromised(`ws://localhost:${port}/`, {
+    packMessage: data => JSON.stringify(data),
+    unpackMessage: data => JSON.parse(data),
+    attachRequestId: (data, requestId) =>
+      Object.assign({ requestId: requestId }, data),
+    extractRequestId: data => data && data.requestId,
+  });
+
+  try {
+    await sock.open();
+    const requests = [];
+
+    events.forEach(event => {
+      let payloads = [];
+
+      if (event.details.input) {
+        payloads = event.details.input.payloads;
+      }
+
+      if (event.details.result) {
+        payloads = event.details.result.payloads;
+      }
```

Comment:
what are input and result and can they coincide?


```suggestion
      if (event.details.input) {
        payloads = event.details.input.payloads;
      } else if (event.details.result) {
        payloads = event.details.result.payloads;
      }
```

what scenarios can `payloads` be an empty array, and if that happens, what should we warn about?